components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -46,11 +46,43 @@ void eddy_diffusivities_c(Int nlev, Int shcol, Real *obklen, Real *pblh,
 
 void calc_shoc_vertflux_c(Int shcol, Int nlev, Int nlevi, Real *tkh_zi,
 			  Real *dz_zi, Real *invar, Real *vertflux);
+
+void shoc_diag_second_moments_srf_c(Int shcol, Real* wthl, Real* uw, Real* vw, Real* ustar2, Real* wstar);
+
 }
 
 namespace scream {
 namespace shoc {
 
+namespace {
+
+template <size_t N, size_t M>
+void gen_random_data(const std::array<std::pair<Real, Real>, N>& ranges,
```

Comment:
Please use the randomize() method off of the base class. If you need a random range other than 0->1, I'm sure you can figure how to do this by adding some optional arguments to randomize.